ui/component/or-rules/src/json-viewer/or-rule-asset-query.ts

```diff
@@ -277,6 +279,11 @@ export class OrRuleAssetQuery extends translate(i18next)(LitElement) {
         // @ts-ignore
         const value = valuePredicate ? valuePredicate.value : undefined;
 
+        // Prepare GeoJsonConfig from predicate.geoJSON when needed
+        const geoJsonConfig = valuePredicate.predicateType === "geojson" && (valuePredicate as any).geoJSON
+            ? (() => { try { return { source: JSON.parse((valuePredicate as any).geoJSON), layers: [] }; } catch { return undefined; } })()
+            : undefined;
```

Comment:
I would put this inside the `case "geojson" { }` clause, since this variable is only used there.